util/eventbus/client.go

```diff
@@ -47,6 +49,16 @@ func (c *Client) Close() {
 	}
 }
 
+func (c *Client) snapshotSubscribeQueue() []DeliveredEvent {
+	c.mu.Lock()
```

Comment:
Maybe add the nil check to the `snapshotQueue` method itself so we don't have to do this lock dance?

Reply:
Done, and refactored the subscriber state getting slightly, because in subsequent hacking I found repeated need for "get subscribeState but don't create one if it doesn't already exist".